great_tables/_formats.py

```diff
@@ -663,112 +778,175 @@ def fmt_scientific(
 
+        if small_pos:
+            # If the value is small enough to not require the (x 10^n) notation, then
+            # the formatted value is based on only the `m_part`
+            x_formatted = m_part
         else:
-            # Define the exponent string based on the `exp_style` that's not the default
-            # value of 'x10n'
+            # Get the set of exponent marks, which are used to decorate the `n_part`
+            exp_marks = _context_exp_marks(context=context)
 
-            exp_str = _context_exp_str(exp_style=exp_style)
+            # Create the formatted string based on `exp_marks` and the two `sci_parts`
+            x_formatted = m_part + exp_marks[0] + n_part + exp_marks[1]
 
-            n_min_width = 1 if _str_detect(exp_style, r"^[a-zA-Z]1$") else 2
+    else:
+        # Define the exponent string based on the `exp_style` that's not the default
+        # value of 'x10n'
 
-            # The `n_part` will be extracted here and it must be padded to
-            # the defined minimum number of decimal places
-            if _str_detect(n_part, "-"):
-                n_part = _str_replace(n_part, "-", "")
-                n_part = n_part.ljust(n_min_width, "0")
-                n_part = "-" + n_part
-            else:
-                n_part = n_part.ljust(n_min_width, "0")
-                if force_sign_n:
-                    n_part = "+" + n_part
+        exp_str = _context_exp_str(exp_style=exp_style)
 
-            # Implement minus sign replacement for `m_part` and `n_part`
-            m_part = _replace_minus(m_part, minus_mark=minus_mark)
-            n_part = _replace_minus(n_part, minus_mark=minus_mark)
+        n_min_width = 1 if _str_detect(exp_style, r"^[a-zA-Z]1$") else 2
 
-            x_formatted = m_part + exp_str + n_part
+        # The `n_part` will be extracted here and it must be padded to
+        # the defined minimum number of decimal places
+        if _str_detect(n_part, "-"):
+            n_part = _str_replace(n_part, "-", "")
+            n_part = n_part.ljust(n_min_width, "0")
+            n_part = "-" + n_part
+        else:
+            n_part = n_part.ljust(n_min_width, "0")
+            if force_sign_n:
+                n_part = "+" + n_part
 
-        # Use a supplied pattern specification to decorate the formatted value
-        if pattern != "{x}":
-            x_formatted = pattern.replace("{x}", x_formatted)
+        # Implement minus sign replacement for `m_part` and `n_part`
+        m_part = _replace_minus(m_part, minus_mark=minus_mark)
+        n_part = _replace_minus(n_part, minus_mark=minus_mark)
 
-        return x_formatted
+        x_formatted = m_part + exp_str + n_part
+
+    # Use a supplied pattern specification to decorate the formatted value
+    if pattern != "{x}":
+
+        # Escape LaTeX special characters from literals in the pattern
+        if context == "latex":
+            pattern = escape_pattern_str_latex(pattern_str=pattern)
```

Comment:
Using `escape_pattern_str_latex()` as before to escape the `pattern` specifically for LaTeX.